### PR TITLE
Only show state question for U.S. students in new sign-up flow

### PR DESCRIPTION
**Pull request by TurnerRiley**
Currently the state question is shown to students in the new sign-up flow regardless of their location. With this PR, it will only show when the student is detected in the U.S.

### Currently shows regardless
oldShowsStateRegardless

### Now does not show if not in the U.S.
nonUs

## Links
Jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-2362)

## Testing story
Local testing and added a unit test.

### CODE REVIEW

File: dashboard/app/controllers/registrations_controller.rb [hunk 1]
```diff
@@ -84,6 +84,10 @@ def finish_student_account
       {value: age.to_s, text: age.to_s}
     end
 
+    # Get the request location
+    location = Geocoder.search(request.ip).try(:first)
+    country_code = location&.country_code.to_s.upcase
+    @us_ip = ['US', 'RD'].include?(country_code)
```
Comment (TurnerRiley) [inline dashboard/app/controllers/registrations_controller.rb:90]: This is copy-pasted from the `finish_teacher_account` method below

**Review by kelbyhawn — APPROVED** [review]
🎉

File: apps/test/unit/signUpFlow/FinishStudentAccountTest.tsx [hunk 2]
```diff
@@ -96,6 +97,13 @@ describe('FinishStudentAccount', () => {
     screen.getByText(locale.go_to_my_account());
   });
 
+  it('does not render state question if user is not detected in the U.S.', () => {
```
Comment (bethanyaconnor) [inline apps/test/unit/signUpFlow/FinishStudentAccountTest.tsx:100]: is it worth adding a test that US state isn't required for non-US IPs?

**Review by bethanyaconnor — APPROVED** [review]
One test suggestion but definitely optional/not blocking!

File: apps/test/unit/signUpFlow/FinishStudentAccountTest.tsx (hunk 2 above)
Comment (TurnerRiley) [reply apps/test/unit/signUpFlow/FinishStudentAccountTest.tsx:100]: I tried to cover that by also checking that the error message doesn't show up. But if this doesn't feel sufficient I can definitely add tests to this and other questions to make sure the user can click the button without selecting a state (or whichever field the test is about)!

Comment (bethanyaconnor) [reply apps/test/unit/signUpFlow/FinishStudentAccountTest.tsx:100]: Does the error message show up if the user is not in the US and hasn't done any other action?

Comment (bethanyaconnor) [reply apps/test/unit/signUpFlow/FinishStudentAccountTest.tsx:100]: Said another way, if you changed line 101 to be `renderDefault(true)` and removed line 103, would this test fail?

Comment (TurnerRiley) [reply apps/test/unit/signUpFlow/FinishStudentAccountTest.tsx:100]: Oh I see what you're saying gotcha. If the user is not in the U.S. the error message should never show up, and in the case of doing your suggestion the test would succeed if the user takes no action. I'll add some tests!

Comment (TurnerRiley) [reply apps/test/unit/signUpFlow/FinishStudentAccountTest.tsx:100]: Adding that sort of test will also prompt the addition of tests to both this file and the teacher side of things so I'm going to make a small ticket as a followup PR!

**Pull request merged**